### PR TITLE
[ip6] add check before adding or removing the unsecure port

### PR DESCRIPTION
**Pull request by librasungirl**

### CODE REVIEW

File: include/openthread/ip6.h [hunk 1]
```diff
@@ -418,8 +418,9 @@ otError otIp6Send(otInstance *aInstance, otMessage *aMessage);
  * @param[in]  aInstance A pointer to an OpenThread instance.
  * @param[in]  aPort     The port value.
  *
- * @retval OT_ERROR_NONE     The port was successfully added to the allowed unsecure port list.
- * @retval OT_ERROR_NO_BUFS  The unsecure port list is full.
+ * @retval OT_ERROR_NONE         The port was successfully added to the allowed unsecure port list.
+ * @retval OT_ERROR_INVALID_ARGS The port is invalid (e.g. the initial value 0).
```
Comment (jwhui) [inline include/openthread/ip6.h:422]:
```suggestion
 * @retval OT_ERROR_INVALID_ARGS The port is invalid (value 0 is reserved).
```

Comment (jwhui) [inline include/openthread/ip6.h:422]:
```suggestion
 * @retval OT_ERROR_INVALID_ARGS The port is invalid (value 0 is reserved for internal use).
```

File: include/openthread/ip6.h [hunk 2]
```diff
@@ -434,8 +435,9 @@ otError otIp6AddUnsecurePort(otInstance *aInstance, uint16_t aPort);
  * @param[in]  aInstance A pointer to an OpenThread instance.
  * @param[in]  aPort     The port value.
  *
- * @retval OT_ERROR_NONE       The port was successfully removed from the allowed unsecure port list.
- * @retval OT_ERROR_NOT_FOUND  The port was not found in the unsecure port list.
+ * @retval OT_ERROR_NONE         The port was successfully removed from the allowed unsecure port list.
+ * @retval OT_ERROR_INVALID_ARGS The port is invalid (e.g. the initial value 0).
```
Comment (jwhui) [inline include/openthread/ip6.h:439]:
```suggestion
 * @retval OT_ERROR_INVALID_ARGS The port is invalid (value 0 is reserved for internal use).
```

File: src/core/net/ip6_filter.hpp [hunk 3]
```diff
@@ -80,8 +80,9 @@ class Filter
      *
      * @param[in]  aPort  The port value.
      *
-     * @retval OT_ERROR_NONE     The port was successfully added to the allowed unsecure port list.
-     * @retval OT_ERROR_NO_BUFS  The unsecure port list is full.
+     * @retval OT_ERROR_NONE         The port was successfully added to the allowed unsecure port list.
+     * @retval OT_ERROR_INVALID_ARGS The port is invalid (e.g. the initial value 0).
```
Comment (librasungirl) [inline src/core/net/ip6_filter.hpp:84]:
```suggestion
     * @retval OT_ERROR_INVALID_ARGS The port is invalid (value 0 is reserved for internal use).
```

File: src/core/net/ip6_filter.hpp [hunk 4]
```diff
@@ -91,8 +92,9 @@ class Filter
      *
      * @param[in]  aPort  The port value.
      *
-     * @retval OT_ERROR_NONE       The port was successfully removed from the allowed unsecure port list.
-     * @retval OT_ERROR_NOT_FOUND  The port was not found in the unsecure port list.
+     * @retval OT_ERROR_NONE         The port was successfully removed from the allowed unsecure port list.
+     * @retval OT_ERROR_INVALID_ARGS The port is invalid (e.g. the initial value 0).
```
Comment (librasungirl) [inline src/core/net/ip6_filter.hpp:96]:
```suggestion
     * @retval OT_ERROR_INVALID_ARGS The port is invalid (value 0 is reserved for internal use).
```

File: src/core/net/ip6_filter.cpp [hunk 5]
```diff
@@ -124,6 +124,8 @@ otError Filter::AddUnsecurePort(uint16_t aPort)
 {
     otError error = OT_ERROR_NONE;
 
+    VerifyOrExit(aPort != 0, error = OT_ERROR_INVALID_ARGS);
```
Comment (bukepo) [inline src/core/net/ip6_filter.cpp:127]: Assert here and move the check to api file?

Comment (librasungirl) [reply src/core/net/ip6_filter.cpp:127]: @bukepo  agree it might be more reasonable. feel free to submit a separate PR to enhance it. thanks.

**Pull request merged**